crates/subspace-networking/src/utils.rs

```diff
@@ -93,3 +95,16 @@ pub(crate) fn convert_multiaddresses(addresses: Vec<Multiaddr>) -> Vec<PeerAddre
         })
         .collect()
 }
+
+pub trait ToMultihash {
+    fn to_multihash(&self) -> Multihash;
+}
+
+impl ToMultihash for PieceIndexHash {
+    fn to_multihash(&self) -> Multihash {
+        libp2p::multihash::MultihashDigest::digest(
+            &Code::Identity,
+            &U256::from(*self).to_be_bytes(),
```

Comment:
As I explained in previous PR, there is no point to convert `PieceIndexHash` to `U256` first, you can use it's `AsRef<[u8]>` directly. instead:
```suggestion
            self.as_ref(),
```